robot_motion_planning_summary.tex

```diff
@@ -31,147 +32,457 @@ \section{Introduction}
+\textbf{Configuration:} The smallest number of real-valued coordinates to fully describe configurations of a robot. For example, a mobile robot moves in directions $x,y$ and can turn its yaw heading $\theta$ - a configuration of this robot could be $\mathbold{c} = (1,3,30^\circ)^\mathrm{T}$. \\
+\textbf{Configuration space (\textit{C-space}):} The space of all possible configurations, that a robot can have in a certain workspace. \\
+\textbf{Dimension of the C-space/DoF:} Minimum number of parameters needed to specify the configuration. Because there can also be angles in the topology of the C-space, it is usually not a Cartesian space (e.g. $2\pi = 4\pi$, cyclic repetition along angle axis). \\
+\textbf{Common robot models:} Single points, single rigid bodies and multiple rigid bodies with joints. \\
+\textbf{Non-holonomic robot:} Robot, that cannot move freely on the
+configuration space manifold because it is constrained in some way. For example, a car cannot drive sideways. \\
+\textbf{Path:} Continuous curve in configuration space $C$ connecting two configurations $q$ and $q'$:
+$\tau: s \in [0,1] \rightarrow \tau(s) \in C$ s.t. $\tau(0) = q$ and $\tau(1) = q'$\\
+\textbf{Trajectory:} A trajectory is a path parametrized by time:
+$\tau: t \in [0, T] \rightarrow \tau(t) \in C$.
+Possible constraints: smoothness, minimum length, minimum time, ...
+
+\subsection{Calculate DoF}
+\textbf{Calculate DoF of a single body:}\\
+DoF = total DoF of 3 fixed points - number of constraints (of the 3 fixed points, e.g. distances between points). \\
+\textit{Alternative:} Find DoF of one point, go the next one and apply constraints, go to the third point and apply constraints. \par
+
+\textbf{Calculate DoF for general robots:}
+\begin{itemize}
+	\item Open chains: add the DoF of each joint.
+	\item Closed chains:\\
+	$\text{DoF} = N(k-1) - \sum_{i=1}^{n} (N-f_i) = N(k-n-1) + \sum_{i=1}^{n} f_i$ \\
+	– $N=6$ for 3D, $N=3$ for 2D\\
+	– $k =$ number of links (including ground link)\\
+	– $n =$ number of joints\\
+	– $f_i =$ DoF of the $i^{th}$ joint
+\end{itemize}
 
-A configuration describes the state of a robot as a combination of different configurations of its degrees of freedom. For example, a mobile robot moves in directions $x,y$ and can turn its yaw heading $\theta$ - a configuration of this robot could be $c = (1,3,30^\circ)^\mathrm{T}$ The configuration space (\textit{C-space}) is the space of all possible configurations, that a robot can have in a certain workspace.
-
-The dimension of the C-space is equal to the degrees of freedom of the robot (minimum number of parameters needed to specify the configuration). Because there can also be angles in the topology of the C-space, it is usually not a cartesian space (e.g. $2\pi = 4\pi$, cyclic repetition along angle axis).
+\subsection{Parametrizations of configurations}
+Configuration: $\mathbold{q}=(position, orientation)=(x,y,z,\cdot)$.
+There exist different ways how to parametrize orientations:
```

Comment:
Angle-axis parameterization could be also added, I don't know if this is relevant for this lecture.

Reply:
I added it.